src/SSHDebugPS/Docker/DockerContainerInstance.cs

```diff
@@ -13,20 +13,19 @@ public class DockerContainerInstance : ContainerInstance
         /// <summary>
         /// Create a DockerContainerInstance from the results of docker ps in JSON format
         /// </summary>
-        public static DockerContainerInstance Create(string json)
+        public static bool TryCreate(string json, out DockerContainerInstance instance)
         {
+            instance = null;
             try
             {
                 JObject obj = JObject.Parse(json);
-                var instance = obj.ToObject<DockerContainerInstance>();
-                if (instance != null)
-                    return instance;
+                instance = obj.ToObject<DockerContainerInstance>();
             }
             catch (Exception e)
             {
                 Debug.Fail(e.ToString());
             }
-            return null;
+            return instance != null;
```

Comment:
should add some logging or assertion if it comes back null

Reply:
And find some way to let the user know that it can't be parsed. Might be good to add some telemetry too.